code/modules/atmospherics/environmental/LINDA_system.dm

```diff
@@ -67,18 +67,17 @@
 		var/turf/T = get_step(src, direction)
 		if(!istype(T))
 			continue
-		var/counterdir = get_dir(T, src)
 		if(CanAtmosPass(T))
 			atmos_adjacent_turfs_amount += 1
-			atmos_adjacent_turfs |= direction
-			if(!(T.atmos_adjacent_turfs & counterdir))
+			atmos_adjacent_turfs |= T
+			if(!(src in T.atmos_adjacent_turfs))

```

Comment:
For reference, I suspect a faster way would be to do T.atmos_adjecent_turfs |= src then T.atmos_adjectent_turfs_amount = T.atmos_adjecent_turfs.len